.github/workflows/api-merge.yml

```diff
@@ -0,0 +1,72 @@
+name: API Merge
+
+on:
+  merge_group:
+    types: [checks_requested]
+
+defaults:
+  run:
+    working-directory: api
+
+jobs:
+  test:
+    runs-on: General-Purpose-8c-Runner
```

Comment:
What's this? Why is it different to our other runners?

Reply:
The goal of this PR is meanly to test and monitor how `testmon` and `merge-queue` are working together.